core/math/a_star.cpp

```diff
@@ -97,11 +97,14 @@ void AStar::remove_point(int p_id) {
 
 	Point *p = points[p_id];
 
-	for (Set<Point *>::Element *E = p->neighbours.front(); E; E = E->next()) {
-
-		Segment s(p_id, E->get()->id);
-		segments.erase(s);
-		E->get()->neighbours.erase(p);
+	Map<int, Point *>::Element* PE = points.front();
```

Comment:
Sorry I missed another style issue when reviewing, see the diff here: https://travis-ci.org/godotengine/godot/jobs/494899675
To prevent such style issues before committing, see [Code style guidelines](http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html).